engine/src/cubos/engine/splitscreen/plugin.cpp

```diff
@@ -0,0 +1,71 @@
+#include <glm/glm.hpp>
+
+#include <cubos/engine/renderer/renderer.hpp>
+#include <cubos/engine/renderer/viewport.hpp>
+#include <cubos/engine/splitscreen/plugin.hpp>
+#include <cubos/engine/splitscreen/splitscreen.hpp>
+#include <cubos/engine/transform/local_to_world.hpp>
+
+using cubos::core::ecs::Commands;
+using cubos::core::ecs::OptWrite;
+using cubos::core::ecs::Query;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+
+using namespace cubos::engine;
+
+static void splitViewport(Commands commands, Write<Renderer> renderer, Read<ActiveCameras> activeCameras,
+                          Query<Read<LocalToWorld>, Write<Camera>, OptWrite<Viewport>> query)
+{
+    glm::ivec2 positions[4];
+    glm::ivec2 sizes[4];
+    int cameraCount = 0;
+
+    for (int i = 0; i < 4; ++i) // NOLINT(modernize-loop-convert)
+    {
+        if (activeCameras->entities[i].isNull())
+        {
+            continue;
+        }
+
+        if (auto components = query[activeCameras->entities[i]])
+        {
+            cameraCount += 1;
+        }
+    }
+
+    setViewportCameras({0, 0}, (*renderer)->size(), cameraCount, positions, sizes);
+
+    for (int i = 0; i < 4; ++i) // NOLINT(modernize-loop-convert)
+    {
+        if (activeCameras->entities[i].isNull())
+        {
+            continue;
+        }
+
+        if (auto components = query[activeCameras->entities[i]])
+        {
+            auto [localToWorld, camera, viewport] = *components;
+            if (!viewport.operator bool())
```

Comment:
```suggestion
            if (!viewport)
```
The `operator bool()` method is defining a cast to bool for the `OptRead` type. You can treat it as if it was boolean :)